client/state/receipts/assembler.ts

```diff
@@ -49,21 +57,19 @@ export function createReceiptObject( data ) {
 /**
  * Purchases are of the format { [siteId]: [ { productId: ... } ] }
  * so we need to flatten them to get a list of purchases
- *
- * @param {import('./types').RawReceiptPurchases | Array<void>} purchases keyed by siteId { [siteId]: [ { productId: ... } ] }
- * @returns {Array<import('./types').RawReceiptPurchase>} of product objects [ { productId: ... }, ... ]
  */
-function flattenPurchases( purchases ) {
-	return flatten( Object.values( purchases ) );
+function flattenPurchases(
+	purchases: RawReceiptPurchases | Array< void >
+): Array< RawReceiptPurchase > {
+	return Object.values( purchases ).flat();
```

Comment:
This is technically doing an `any[] => Array< RawReceiptPurchase>` cast under the hood. If you filter out `void[]` you wont' run into that problem.

Reply:
How do you filter that out? (And did I correctly type that `[]` is `Array<void>`?) Something like this?

```suggestion
        if ( purchases === [] ) {
          return [];
         }
	return Object.values( purchases ).flat();
```

Reply:
I think you can do the exact same as above `Array.isArray( purchases ) ? {} : purchases` or similar.

Reply:
I did check and `Object.values( [] )` is not an error (since arrays are objects). So would doing this be just for further type safety?

Reply:
Yeah, it's just to avoid `any`

Reply:
Done in https://github.com/Automattic/wp-calypso/pull/69638